diff --git a/core/lib/Drupal/Core/Render/MainContent/ModalRenderer.php b/core/lib/Drupal/Core/Render/MainContent/ModalRenderer.php
index 60a495122d..a279f37dc9 100644
--- a/core/lib/Drupal/Core/Render/MainContent/ModalRenderer.php
+++ b/core/lib/Drupal/Core/Render/MainContent/ModalRenderer.php
@@ -28,6 +28,9 @@ public function renderResponse(array $main_content, Request $request, RouteMatch
 
     // If the main content doesn't provide a title, use the title resolver.
     $title = isset($main_content['#title']) ? $main_content['#title'] : $this->titleResolver->getTitle($request, $route_match->getRouteObject());
+    if(is_array($title)){
+      $title = drupal_render_root($title);
+    }
 
     // Determine the title: use the title provided by the main content if any,
     // otherwise get it from the routing information.
